test/common/ssl/ssl_socket_test.cc

```diff
@@ -3536,11 +3536,14 @@ class SslReadBufferLimitTest : public SslSocketTest {
 
       // Incredibly contrived way of making sure that the write buffer has an empty chain in it.
       if (reserve_write_space) {
-        Buffer::RawSlice iovecs[2];
-        EXPECT_EQ(2UL, data.reserve(16384, iovecs, 2));
-        iovecs[0].len_ = 0;
-        iovecs[1].len_ = 0;
-        data.commit(iovecs, 2);
+        Buffer::RawSlice iovec1[1];
```

Comment:
why does this test need to change?

Reply:
This test depended on an implementation detail of `evbuffer`: that particular call to `reserve` resulted in the creation of two separate slices inside the buffer. The new `OwnedImpl` implementation satisfies the reservation with a single slice in this case. One of the `TODO` items in `buffer_impl.cc` will probably result in the new implementation using two slices in this case in the future, to match how `evbuffer` works today, but I still wanted to make the test less dependent on implementation details.

Reply:
is that still true? or have you fixed this now?

Reply:
The original code here will now work once again, so I'll revert to that.